pkg/apps/controller/deployer/deployer_controller_test.go

```diff
@@ -1075,3 +1082,76 @@ func TestDeployerCustomLabelsAndAnnotations(t *testing.T) {
 		expectMapContains(t, podTemplate.Annotations, test.annotations, "annotations")
 	}
 }
+
+func TestMakeDeployerPod(t *testing.T) {
+	client := &fake.Clientset{}
+	controller := okDeploymentController(client, nil, nil, true, v1.PodUnknown)
+	config := deploytest.OkDeploymentConfig(1)
+	deployment, _ := deployutil.MakeDeploymentV1(config, codec)
+	container := controller.makeDeployerContainer(&config.Spec.Strategy)
+	container.Resources = deployutil.CopyApiResourcesToV1Resources(&config.Spec.Strategy.Resources)
+	defaultGracePeriod := int64(10)
+	maxDeploymentDurationSeconds := deployapi.MaxDeploymentDurationSeconds
+
+	for i := 1; i <= 25; i++ {
+		seed := rand.Int63()
+		f := fuzzer.FuzzerFor(kapitesting.FuzzerFuncs, rand.NewSource(seed), kapi.Codecs)
+		f.Funcs(
+			func(p *kapiv1.PodTemplateSpec, c fuzz.Continue) {
+				c.FuzzNoCustom(p)
+				p.Spec.InitContainers = nil
+				containerCopy := container.DeepCopy()
+
+				// These are specific for deployer pod container:
+				p.Spec.Containers = []kapiv1.Container{*containerCopy}
+				p.Spec.Containers[0].Name = "deployment"
+				p.Spec.Containers[0].Command = containerCopy.Command
+				p.Spec.Containers[0].Args = containerCopy.Args
+				p.Spec.Containers[0].Image = containerCopy.Image
+				p.Spec.Containers[0].Env = append(p.Spec.Containers[0].Env, kapiv1.EnvVar{Name: "OPENSHIFT_DEPLOYMENT_NAME", Value: "config-1"})
+				p.Spec.Containers[0].Env = append(p.Spec.Containers[0].Env, kapiv1.EnvVar{Name: "OPENSHIFT_DEPLOYMENT_NAMESPACE", Value: "default"})
+				p.Spec.Containers[0].Resources = containerCopy.Resources
+				p.Spec.Containers[0].ImagePullPolicy = kapiv1.PullIfNotPresent
+
+				// These are hardcoded for deployer pod spec
+				p.Spec.RestartPolicy = kapiv1.RestartPolicyNever
+				p.Spec.TerminationGracePeriodSeconds = &defaultGracePeriod
+				p.Spec.ActiveDeadlineSeconds = &maxDeploymentDurationSeconds
+				p.Spec.ServiceAccountName = "sa:test"
+
+				// FIXME: These are weird or missing. If you get an error below, consider
+				// adding this field into deployer controller or to this list:
```

Comment:
@kargakis @tnozicka @smarterclayton we probably need to decide which fields we want to copy from the deployment config template to deployer pod (some of these are not copied for reason (like Volume...), some of them we should copy? (like NodeName?)..))

Reply:
I don't think any of these should be copied except for volumes only in case of customStrategy otherwise that would be a security hole

Reply:
feels like it should be the cluster admin who gets to choose deployers scheduling, not the user

Reply:
Running pods on the users behalf is always fraught - we have to be careful what we allow.  We have gradually given users more control over deployer pods including location of scheduling, but our current API won’t scale by one off adding fields.

Toleration improve scheduling (give the pod more options) in most cases, so it’s probably correct to copy them.  However, we might have to filter some of the conditions to be safe.  For now, I’m a bit concerned about adding them without some design time.

Reply:
@smarterclayton i'm fine keeping this test as it is (it will alert us if new fields are added to PodSpec) with a follow up issue where we can discuss what fields we want to copy.